lib/ruby/stdlib/jruby/core_ext/class.rb

```diff
@@ -153,22 +153,31 @@ def add_method_signature(name, classes)
     self_r.add_method_signature(name, types.to_java(JClass))
   end
 
-  def java_field(signature)
-    signature = signature.to_s
-
-    signature = signature.split(/\s/)
+  ##
+  # java_field will take the argument and create a java field as defined
+  # with the name, the Java type, and the annotation signatures.
+  # 
+  # :call-seq:
+  #   java_field '@FXML int foo'
+  #   java_field 'org.foo.Bar bar'
+  def java_field(signature_source)
+    signature = JRuby::JavaSignature.parse "#{signature_source}()"
 
-    raise "Java Field must be specified as a string with the format <Type Name>" if signature.size != 2
+    add_field_signature signature.name, signature.return_type
 
-    type, name = signature
-    add_field_signature(name, type)
+    annotations = signature.annotations
+    add_field_annotation signature.name, annotations if annotations
   end
 
+  def java_annotation(anno)
+    STDERR.puts "java_annotation is deprecated. Use java_signature '@#{anno} ...' instead. Called from: #{caller.first}"
```

Comment:
minor: we probably just want to use `warn` or some kind of warnings on the native end.

Reply:
Yeah use Kernel#warn or Warnings#warn so it can be silenced if necessary.